src/tokens/tokens.utils.ts

```diff
@@ -115,6 +120,78 @@ export async function getCachedTokenBalances(
   return undefined;
 }
 
+export async function getFullToken(chain: Chain, tokenAddr: Token['address']): Promise<TokenFull | null> {
+  const fullTokenMap = await getFullTokens(chain, [tokenAddr]);
+  return fullTokenMap[tokenAddr] || null;
+}
+
+export async function getFullTokens(chain: Chain, tokensAddr: Token['address'][]): Promise<TokenFullMap> {
+  const cachedTokens = await getCachedTokesInfo(tokensAddr);
+  const cachedTokensAddr = cachedTokens.map((token) => token.address);
+
+  const tokensCacheMissMatch = tokensAddr.filter((addr) => !cachedTokensAddr.includes(addr));
+
+  if (tokensCacheMissMatch.length === 0) return mergeTokensFullData(cachedTokens);
+
+  const sdk = await chain.getSdk();
+  let tokensInfo: Token[] = [];
+  try {
+    const sdkTokensInfo = await sdk.tokens.loadTokens(tokensCacheMissMatch);
+    tokensInfo = Object.values(sdkTokensInfo);
+  } catch (e) {
+    console.warn(`Faild to load tokens from chain node ${e}`);
+  }
+
+  if (tokensInfo.length > 0) await cacheTokensInfo(tokensInfo);
+
+  const tokensList = tokensInfo.concat(cachedTokens);
+
+  return mergeTokensFullData(tokensList);
+}
+
+export async function getCachedTokesInfo(tokensAddr: Token['address'][]): Promise<Token[]> {
+  const mapper = getDataMapper();
+
+  const tokensInfo: Token[] = [];
+
+  for (const addr of tokensAddr) {
+    try {
+      tokensInfo.push(await mapper.get(Object.assign(new TokenInformationSnapshot(), { address: addr })));
```

Comment:
if we are getting arrays let's try to use batchGet

https://github.com/Badger-Finance/badger-api/blob/main/src/indexers/boosts-indexer.ts#L40-L48